### ISSUE TITLE
Higher-order gradient

### ISSUE THREAD

**Issue by ThibaultGROUEIX** [mabaorui/NeuralPull#6]
Hi @mabaorui,

Congrats again and sorry for the repeated queries. I am still trying to figure out a few aspects of the method and am thus opening various discussions, I hope that is ok.

I wonder if you use higher-order gradients to backprop in equation 1 of the main paper.
I noticed that the pytorch version [does not](https://github.com/bearprin/neuralpull-pytorch/blob/master/train.py#L78) use higher order gradients. And I have hardly ever used Tensorflow in my life, so it is hard for me to understand the official code.

Beside the answer, I wonder what are the pros and cons of using higher-order gradients in this equation, and how you made this design decision,

Thanks a lot,
Thibault
